src/fmt.rs

```diff
@@ -154,20 +162,29 @@ where
 
 /// Indicates a format which corresponds to a Rust slice type,
 /// representing exactly the same invariants.
+/// 
+/// # Safety
+/// Inherits safety constraints from [`Format`]
 pub unsafe trait SliceFormat: Format + Sized {
     type Slice: ?Sized + Slice;
 }
 
 /// Indicates a format which contains characters from Unicode
 /// (all of it, or some proper subset).
+/// 
+/// # Safety
+/// 
+/// Changing format without validation will cause unsoundness.
 pub unsafe trait CharFormat<'a>: Format {
     /// Iterator for characters and their byte indices.
     type Iter: Iterator<Item = (usize, char)>;
 
     /// Iterate over the characters of the string and their byte
     /// indices.
-    ///
-    /// You may assume the buffer is *already validated* for `Format`.
+    /// 
+    /// # Safety 
+    /// 
+    /// It assumes the buffer is *already validated* for `Format`.
```

Comment:
Typical style for safety comments is using language about what the function needs not what the function assumes.

So: "Must be passed a buffer that is valid for Format"